numba/cuda/cudaimpl.py

```diff
@@ -837,27 +846,28 @@ def _generic_array(context, builder, shape, dtype, symbol_name, addrspace,
         lmod = builder.module
 
         # Create global variable in the requested address space
-        gvmem = lmod.add_global_variable(laryty, symbol_name, addrspace)
+        gvmem = cgutils.add_global_variable(lmod, laryty, symbol_name,
+                                            addrspace)
         # Specify alignment to avoid misalignment bug
         align = context.get_abi_sizeof(lldtype)
         # Alignment is required to be a power of 2 for shared memory. If it is
         # not a power of 2 (e.g. for a Record array) then round up accordingly.
         gvmem.align = 1 << (align - 1 ).bit_length()
 
         if dynamic_smem:
-            gvmem.linkage = lc.LINKAGE_EXTERNAL
+            gvmem.linkage = 'external'
```

Comment:
Wonder if the `enum` in `llvmlite` ought to be moved out of `llvmpy` and into the `llvmlite` API such that reliance on strings is reduced? (Not for this PR!).

Reply:
Maybe... I assumed that the use of strings instead of the enum was a deliberate design decision to permit flexibility in the linkage types for the user of llvmlite, but it's probably flexibility that's turned out to be un-needed (unless it's being used by external non-Numba users).